src/hooks/protections/receive_net_message.cpp

```diff
@@ -98,227 +321,439 @@ namespace big
 			}
 		}
 
-		if (!get_msg_type(msgType, buffer))
-			return g_hooking->get_original<hooks::receive_net_message>()(netConnectionManager, a2, frame);
+		int sec_id = 0;
 
+		if (frame)
+			sec_id = frame->m_security_id;
+		else
+		{
+			if (auto cxn = g_pointers->m_gta.m_get_connection_peer(net_cxn_mgr, error->m_peer_id))
+				sec_id = cxn->m_security_id;
+		}
 
-		if (msgType == rage::eNetMessage::MsgRoamingJoinBubbleAck)
+		auto peer = g_pointers->m_gta.m_get_peer_by_security_id(sec_id); // shouldn't be null in most cases, contains unspoofable data
+
+		if (error && msgType != rage::eNetMessage::MsgJoinResponse)
+		{
+			if (peer)
+				LOGF(stream::net_messages, WARNING, "Received an error packet that isn't MsgJoinResponse from {}", peer->m_info.name);
 			return true;
+		}
+
+		if (g.debug.logs.packet_logs)
+		{
+			log_net_message(msgType, buffer, event, peer);
+		}
 
-		if (player)
+		switch (msgType)
 		{
-			switch (msgType)
+		case rage::eNetMessage::MsgScriptJoin:
+		{
+			CGameScriptId script;
+			script_id_deserialize(script, buffer);
+
+			if (player && handle_block_script(player, script, event->m_msg_id))
 			{
-			case rage::eNetMessage::MsgTextMessage:
-			case rage::eNetMessage::MsgTextMessage2:
+				LOGF(stream::net_messages, WARNING, "Denying script request from {} (hash={:X}, instance={})", player->get_name(), script.m_hash, script.m_instance_id);
+				return true;
+			}
+
+			break;
+		}
+		case rage::eNetMessage::MsgScriptHostRequest:
```

Comment:
is this event still working? or we need to move "freemode"_J logic into `rage::eNetMessage::MsgScriptJoin`?